### PR TITLE
[Actions] People Data Labs

### LINKED ISSUE #2893: [Actions] People Data Labs
**Is there a specific app this action is for?**
`people_data_labs`

**Please provide a link to the relevant API docs for the specific service / operation.**
https://docs.peopledatalabs.com/docs/enrichment-api

### Actions
- Enrich company 
- Enrich person

### PR DESCRIPTION
**Pull request by vellames-turing**
Closes #2893

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/people_data_labs/actions/enrich-company/enrich-company.mjs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+      country: this.country,
+      street_address: this.streetAddress,
+      postal_code: this.postalCode,
+      data_include: this.dataInclude,
+      pretty: this.pretty,
+      min_likelihood: this.minLikelihood,
+    };
+
+    const res = await this.app.enrichCompany(params)
+    if (!res) {
```
Comment (lcaresia) [inline components/people_data_labs/actions/enrich-company/enrich-company.mjs:108]: If don't have results, is not better just return the empty result for the user?

File: components/people_data_labs/actions/enrich-person/enrich-person.mjs [hunk 2]
```diff
@@ -1,149 +1,171 @@
...
     },
-    min_likelihood: {
-      type: "string",
-      description: "The minimum likelihood score a response must have to return a 200.",
-      optional: true,
-    },
     required: {
+      label: "Required",
       type: "string",
       description: "Parameter specifying the fields and data points a response must have to return a 200.  See [docs](https://docs.peopledatalabs.com/docs/required-parameter).",
```
Comment (lcaresia) [inline components/people_data_labs/actions/enrich-person/enrich-person.mjs:122]:
```suggestion
      description: "Parameter specifying the fields and data points a response must have to return a 200. See [docs](https://docs.peopledatalabs.com/docs/required-parameter).",
```

File: components/people_data_labs/actions/enrich-person/enrich-person.mjs [hunk 3]
```diff
@@ -1,149 +1,171 @@
...
       optional: true,
     },
     profile: {
+      label: "Profile",
       type: "string",
       description: "A social profile the person has used. [List of available social profiles](https://docs.peopledatalabs.com/docs/social-networks).",
       optional: true,
     },
     lid: {
+      label: "Linkedin Id",
```
Comment (lcaresia) [inline components/people_data_labs/actions/enrich-person/enrich-person.mjs:108]:
```suggestion
      label: "Linkedin ID",
```

File: components/people_data_labs/people_data_labs.app.mjs [hunk 4]
```diff
@@ -1,11 +1,71 @@
...
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    _getBaseUrl() {
+      return "https://api.peopledatalabs.com/v5";
+    },
+    _getHeaders() {
+      return {
+        "Content-Type": "application/json",
+        "X-API-Key": this.$auth.api_key,
```
Comment (lcaresia) [inline components/people_data_labs/people_data_labs.app.mjs:27]: Create and call a getter here.

E.g. `this._apiKey()`

File: components/people_data_labs/people_data_labs.app.mjs [hunk 5]
```diff
@@ -1,11 +1,71 @@
...
+      };
+    },
+    _getAxiosParams(opts = {}) {
+      const res = {
+        ...opts,
+        url: this._getBaseUrl() + opts.path,
+        headers: this._getHeaders(),
+      };
+      delete res.path;
+      console.log(res);
```
Comment (lcaresia) [inline components/people_data_labs/people_data_labs.app.mjs:37]:
```suggestion
```

File: components/people_data_labs/people_data_labs.app.mjs [hunk 6]
```diff
@@ -1,11 +1,71 @@
...
+      return res;
+    },
+    async _makeRequest(opts = {}) {
+      try {
+        const axiosParams = this._getAxiosParams(opts);
+        return await axios(this, axiosParams);
+      } catch (err) {
+        if (err)
```
Comment (lcaresia) [inline components/people_data_labs/people_data_labs.app.mjs:45]:
```suggestion
```

File: components/people_data_labs/people_data_labs.app.mjs [hunk 7]
```diff
@@ -1,11 +1,71 @@
...
+        if (err?.response?.data?.error) {
+          if (err.response.data.error.status == 404) {
+            return null;
+          }
+          throw new Error(err.response.data.error.message)
+        }
+        throw err;
+      }
```
Comment (lcaresia) [inline components/people_data_labs/people_data_labs.app.mjs:53]: Don't need validate this.

`@pipedream/platform axios` do it natively.

File: components/people_data_labs/people_data_labs.app.mjs [hunk 8]
```diff
@@ -1,11 +1,71 @@
...
+      
+    },
+    async enrichPerson(params) {
+      return this._makeRequest({
+        method: "get",
```
Comment (lcaresia) [inline components/people_data_labs/people_data_labs.app.mjs:58]: Method `get` is default, you don't need pass this.

File: components/people_data_labs/people_data_labs.app.mjs [hunk 9]
```diff
@@ -1,11 +1,71 @@
...
+        path: "/person/enrich",
+        params
+      })
     },
+    async enrichCompany(params) {
+      return this._makeRequest({
+        method: "get",
```
Comment (lcaresia) [inline components/people_data_labs/people_data_labs.app.mjs:65]: Method `get` is default, you don't need pass this.

**Review by lcaresia — CHANGES_REQUESTED** [review]
Thanks for your contribution.

Just a minor changes.

File: components/people_data_labs/actions/enrich-company/enrich-company.mjs (hunk 1 above)
Comment (vellames-turing) [reply components/people_data_labs/actions/enrich-company/enrich-company.mjs:108]: Normally we return nothing which will be `undefined`

**Pull request merged**